### DISCUSSION TITLE
Configuring proxy_det_header X-Request-Id $request_id in http>server does not take effect

### DISCUSSION THREAD

**Discussion by guanzhaogz** [nginx/nginx#407] [Q&A] [answered]
CentOS 7 Configuring proxy_det_header X-Request-Id $request_id in http>server does not take effect, the same configuration takes effect in Windows. May I ask why this is? Thank you.

http {
    include       mime.types;
    default_type  application/octet-stream;

    sendfile        on;
    #tcp_nopush     on;

    #keepalive_timeout  0;
    keepalive_timeout  65;
    server_tokens off;
    limit_conn_zone $binary_remote_addr zone=one:10m;
    client_header_timeout 60;
    client_body_timeout 60;
    proxy_send_timeout 60;
    proxy_read_timeout 120s;
    send_timeout 60;
    include       vhost/*.conf;

    #gzip  on;

server{
	listen 8085;
	proxy_set_header X-Request-Id $request_id;
	location / {
		proxy_set_header Host $proxy_host;
		proxy_pass http://127.0.0.1:8088;
	}
}
}

**Comment by HanadaLee** [comment]
https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_set_header

```
These directives are inherited from the previous configuration level if and only if there are no proxy_set_header directives defined on the current level.
```

you can use follow config
```
http {
include mime.types;
default_type application/octet-stream;

sendfile        on;
#tcp_nopush     on;

#keepalive_timeout  0;
keepalive_timeout  65;
server_tokens off;
limit_conn_zone $binary_remote_addr zone=one:10m;
client_header_timeout 60;
client_body_timeout 60;
proxy_send_timeout 60;
proxy_read_timeout 120s;
send_timeout 60;
include       vhost/*.conf;

#gzip  on;
server{
listen 8085;
location / {
proxy_set_header X-Request-Id $request_id;
proxy_set_header Host $proxy_host;
proxy_pass http://127.0.0.1:8088/;
}
}
}
```

**Reply by guanzhaogz** [reply]
![image](https://github.com/user-attachments/assets/2f902bdb-cd94-4038-97cb-037a8a68f32d)

server among them

**Reply by HanadaLee** [reply, marked as answer]
These directives are inherited from the previous configuration level if and only if there are no proxy_set_header directives defined on the current level.

If you remove all proxy_set_header directives from location, then the directives on the server will take effect
